### PR TITLE
fix: replace mouseUp with auxClick

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Replace mouseUp with auxClick
- Verified locally on events, will do extra check once deployed as well

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1883 #done

### CODE REVIEW

File: packages/shared/src/components/cards/AdLink.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default function AdLink({ ad, onLinkClick }: AdLinkProps): ReactElement {
       rel="noopener"
       title={ad.description}
       onClick={() => onLinkClick?.(ad)}
-      onMouseUp={(event) => event.button === 1 && onLinkClick?.(ad)}
+      onAuxClick={(event) => event.button === 1 && onLinkClick?.(ad)}
```
Comment (omBratteng) [inline packages/shared/src/components/cards/AdLink.tsx:18]: FYI auxclick is not [supported on Safari](https://caniuse.com/auxclick) 😬

Comment (rebelchris) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Weird, I tested it now and it fires correctly on Safari 🤯
I'll try mobile to check as well

Comment (omBratteng) [reply packages/shared/src/components/cards/AdLink.tsx:18]: What is the intended action to happen?

Comment (rebelchris) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Best example:
Card -> Read article button
Middle click it
Actions to happen:
- Open in new tab
- Track event for go to link

Comment (rebelchris) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Never mind my mobile comment lol 😂

Comment (omBratteng) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Well, with the new iPhone 17 with USB-C you can connect a mouse I think 🤓

Comment (omBratteng) [reply packages/shared/src/components/cards/AdLink.tsx:18]: My local environment has crashed hard right now, so I can't test locally.
Though I think the default action for middle click on macOS is to open links in new tab, probably why it works.

You could try to remove `onAuxClick` and test it in Safari and see if it still works

Comment (rebelchris) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Yep you are totally right, removing onAuxClick will still work on Safari indeed.
It opens new tab + tracks the link

Chrome with it disabled:
Opens new tab + does NOT track the link

With auxClick they both work as intended

**Review by omBratteng — APPROVED** [review]

File: packages/shared/src/components/cards/AdLink.tsx (hunk 1 above)
Comment (omBratteng) [reply packages/shared/src/components/cards/AdLink.tsx:18]: Perfect 😄

**Review by capJavert — APPROVED** [review]
Awesome, can we also replace all separate `onAuxClick` instances with `combinedClicks`?

File: packages/shared/src/components/cards/AdLink.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/components/cards/AdLink.tsx:18]: > My local environment has crashed hard right now, so I can't test locally.
Though I think the default action for middle click on macOS is to open links in new tab, probably why it works.

> You could try to remove onAuxClick and test it in Safari and see if it still works

Good to know, I know we tested `onAuxClick` on firefox and safari and it worked, but did not know about default behaviour 🤘

**Review by capJavert — APPROVED** [review]

**Pull request merged**